internal/driver/custommetadata.go

```diff
@@ -24,19 +24,24 @@ func (onvifClient *OnvifClient) setCustomMetadata(device contract.Device, data [
 		return device, errors.NewCommonEdgeX(errors.KindServerError, "failed to unmarshal the json request body", err)
 	}
 	if len(dataObj) == 0 {
-		return device, errors.NewCommonEdgeX(errors.KindServerError, "no data in PUT command", err)
+		return device, errors.NewCommonEdgeX(errors.KindIOError, "no data in PUT command", err)
```

Comment:
if this was http, I would expect a 400 bad request, so according to edgex, that would map to `errors.KindContractInvalid`

```go
	case http.StatusBadRequest:
		return KindContractInvalid
```